codebuild/bin/grep_simple_mistakes.sh

```diff
@@ -77,6 +77,18 @@ for file in $S2N_FILES_ASSERT_NOTNULL_CHECK; do
   done < <(grep -rnE -A 1 "=\ss2n_stuffer_raw_read\(.*\)" $file)
 done
 
+# Assert that "index" is not a variable name. An "index" function exists in strings.h, and older compilers (<GCC 4.8) 
+# warn if any local variables called "index" are used because they are considered to shadow that declaration. 
+S2N_FILES_ASSERT_VARIABLE_NAME_INDEX=$(find "$PWD" -type f -name "s2n*.[ch]")
+for file in $S2N_FILES_ASSERT_VARIABLE_NAME_INDEX; do
+
+RESULT_VARIABLE_NAME_INDEX=`grep -rn '\(int\|uint[0-9]*_t\|size_t\|bool\|char\) \**index' $file`
```

Comment:
I'm not sure trying to enumerate every possible data type is the right approach here. What happens if you just search for " index " (spaces on either side)?